src/Scientist/Internals/Experiment.cs

```diff
@@ -66,13 +72,30 @@ public ExperimentInstance(string name, Func<Task<T>> control, Func<Task<T>> cand
             return controlResult.Result;
         }
 
+        private bool CompareResults(ExperimentResult controlResult, ExperimentResult candidateResult)
+        {
+            if (_resultComparison != null)
+            {
+                return _resultComparison(controlResult.Result, candidateResult.Result);                
```

Comment:
As I think about this more, I worry taht people supplying a comparer function will forget to handle null checks. In fact, why should we make them?

If exactly one of the results is null and the other is not, then we know to return false.
If both results are null, we return true.

If both results are not null, we then continue and use the result comparers etc.

That way, developers who pass in a `resultComparison` don't need to worry about checking for null. The tricky part is that `T` is not constrained to be a reference type. So we'd probably need to cast to `object` and then do the null checks. That leads to boxing in the case of value types, which sucks, but I think it makes the API easier to use. Thoughts?